data/pools/transactionPool.go

```diff
@@ -582,16 +593,23 @@ func (pool *TransactionPool) addToPendingBlockEvaluatorOnce(txgroup []transactio
 					stats.StopReason = telemetryspec.AssembleBlockFull
 				} else {
 					stats.StopReason = telemetryspec.AssembleBlockTimeout
+					// if the block is not full, it means that the above transaction made it to the block, so we want to add it here.
+					stats.ProcessingTime.AddTransaction(transactionGroupDuration)
```

Comment:
the stats nil check below makes me notice that there isn't one here, but apparently it works out that this code is only reached through paths that set stats

Reply:
I think that stat != nil is always the case when we recompute.

Reply:
yes, confirmed. I'll remove the stat != nil check.